### PR TITLE
Release candidate 3.6.01

### PR DESCRIPTION
**Pull request by ndellingwood**

### CODE REVIEW

**Comment by ndellingwood** [comment]
CI failure with OMP Target , some linking issue:
```
lld: error: /tmp/libkokkoscore-gfx906-7aefc9-gfx906-ac67df.o: symbol _HASHW_/var/jenkins/workspace/Kokkos@2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___454_wrapper has undefined version 2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___454_wrapper

lld: error: /tmp/libkokkoscore-gfx906-7aefc9-gfx906-ac67df.o: symbol _HASHW_/var/jenkins/workspace/Kokkos@2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___456_wrapper has undefined version 2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___456_wrapper

lld: error: /tmp/libkokkoscore-gfx906-7aefc9-gfx906-ac67df.o: symbol _HASHW_/var/jenkins/workspace/Kokkos@2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___508_wrapper has undefined version 2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___508_wrapper

lld: error: /tmp/libkokkoscore-gfx906-7aefc9-gfx906-ac67df.o: symbol _HASHW_/var/jenkins/workspace/Kokkos@2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___510_wrapper has undefined version 2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___510_wrapper

lld: error: /tmp/libkokkoscore-gfx906-7aefc9-gfx906-ac67df.o: symbol _HASHW_/var/jenkins/workspace/Kokkos@2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___534_wrapper has undefined version 2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___534_wrapper

lld: error: /tmp/libkokkoscore-gfx906-7aefc9-gfx906-ac67df.o: symbol _HASHW_/var/jenkins/workspace/Kokkos@2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___536_wrapper has undefined version 2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___536_wrapper

lld: error: /tmp/libkokkoscore-gfx906-7aefc9-gfx906-ac67df.o: symbol _HASHW_/var/jenkins/workspace/Kokkos@2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___580_wrapper has undefined version 2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___580_wrapper

lld: error: /tmp/libkokkoscore-gfx906-7aefc9-gfx906-ac67df.o: symbol _HASHW_/var/jenkins/workspace/Kokkos@2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___582_wrapper has undefined version 2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___582_wrapper

lld: error: /tmp/libkokkoscore-gfx906-7aefc9-gfx906-ac67df.o: symbol _HASHW_/var/jenkins/workspace/Kokkos@2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___616_wrapper has undefined version 2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___616_wrapper

lld: error: /tmp/libkokkoscore-gfx906-7aefc9-gfx906-ac67df.o: symbol _HASHW_/var/jenkins/workspace/Kokkos@2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___618_wrapper has undefined version 2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___618_wrapper

lld: error: /tmp/libkokkoscore-gfx906-7aefc9-gfx906-ac67df.o: symbol _HASHW_/var/jenkins/workspace/Kokkos@2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___650_wrapper has undefined version 2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___650_wrapper

lld: error: /tmp/libkokkoscore-gfx906-7aefc9-gfx906-ac67df.o: symbol _HASHW_/var/jenkins/workspace/Kokkos@2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___652_wrapper has undefined version 2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___652_wrapper

lld: error: /tmp/libkokkoscore-gfx906-7aefc9-gfx906-ac67df.o: symbol _HASHW_/var/jenkins/workspace/Kokkos@2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___686_wrapper has undefined version 2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___686_wrapper

lld: error: /tmp/libkokkoscore-gfx906-7aefc9-gfx906-ac67df.o: symbol _HASHW_/var/jenkins/workspace/Kokkos@2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___688_wrapper has undefined version 2/build/algorithms/unit_tests/openmptarget/TestRandomAndSort_cpp__omp_outlined___688_wrapper

clang-13: error: amdgcn-link command failed with exit code 1 (use -v to see invocation)

...
```

Is this related to a missing update in the PR or is this a CI quirk?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+### Bug Fixes:
+- Fix ScatterMin/ScatterMax to use proper atomics (3.6.01 cherry-pick) [\#5046](https://github.com/kokkos/kokkos/pull/5046)
+- Fix allocating large Views [\#4907](https://github.com/kokkos/kokkos/pull/4907)
+- Fix bounds errors with Kokkos::sort [\#4980](https://github.com/kokkos/kokkos/pull/4980)
+- Fix HIP version when printing the configuration [\#4872](https://github.com/kokkos/kokkos/pull/4872)
+- Fixed _CUDA_ARCH__ to __CUDA_ARCH__ for CUDA LDG [\#4893](https://github.com/kokkos/kokkos/pull/4893)
+- Fixed an incorrect struct initialization [\#5028](https://github.com/kokkos/kokkos/pull/5028)
+- Fix device_id in profiling [\#4997](https://github.com/kokkos/kokkos/pull/4997)
+- Fix racing condition in HIPParallelLaunch [\#5008](https://github.com/kokkos/kokkos/pull/5008)
+- Avoid deprecation warnings with OpenMPExec::validate_partition [\#4982](https://github.com/kokkos/kokkos/pull/4982)
```
Comment (dalg24) [inline CHANGELOG.md:15]:
```suggestion
- Fixed `_CUDA_ARCH__` to `__CUDA_ARCH__` for CUDA LDG [\#4893](https://github.com/kokkos/kokkos/pull/4893)
- Fixed an incorrect struct initialization [\#5028](https://github.com/kokkos/kokkos/pull/5028)
- Fix `device_id` in profiling [\#4997](https://github.com/kokkos/kokkos/pull/4997)
- Fix racing condition in `HIPParallelLaunch` [\#5008](https://github.com/kokkos/kokkos/pull/5008)
- Avoid deprecation warnings with `OpenMPExec::validate_partition` [\#4982](https://github.com/kokkos/kokkos/pull/4982)
```

**Comment by dalg24** [comment]
This OpenMPTarget-ROCm build has been failing on all CI builds, we need to investigate but you can ignore here.

File: CHANGELOG.md (hunk 1 above)
Comment (ndellingwood) [reply CHANGELOG.md:15]: @dalg24 are you able to commit the suggestion directly to release-candidate-3.6.01? Permissions won't allow me to commit directly and would require a PR to the branch before updating here

**Review by crtrott — APPROVED** [review]

**Review by dalg24 — APPROVED** [review]

**Comment by masterleinad** [comment]
This depends on #5122.

**Comment by ndellingwood** [comment]
Is the OpenMPTarget CI failure related to changes in this PR?

```
4: [ RUN      ] openmptarget.unique_token_global

4: /var/jenkins/workspace/Kokkos/core/unit_test/TestUniqueToken.hpp:151: Failure

4: Expected equality of these values:

4:   sum

4:     Which is: 9999990

4:   int64_t(N) * R

4:     Which is: 10000000

4: [  FAILED  ] openmptarget.unique_token_global (212 ms)

```

**Comment by masterleinad** [comment]
> Is the OpenMPTarget CI failure related to changes in this PR?

We have seen this before.

**Review by masterleinad — APPROVED** [review]

**Pull request merged**